app/internal/security/dependancies.py

```diff
@@ -0,0 +1,66 @@
+from typing import Union
+
+from app.dependencies import get_db
+from app.database.models import User
+from app.internal.security.ouath2 import (
+    Depends, Session, check_jwt_token,
+    get_authorization_cookie, HTTP_401_UNAUTHORIZED,
+    HTTPException)
+from starlette.requests import Request
+
+
+async def get_user(
+        db: Session, user_id: int, username: str, path: str) -> User:
+    '''
+    Helper for dependency functions.
+    Recives user details from decrypted jwt token,
+    and check them against the database.
+    returns User base model instance if succeeded,
+    raises HTTPException if fails.
+    '''
+    db_user = await User.get_by_username(db, username=username)
+    if db_user and db_user.id == user_id:
+        return db_user
+    else:
+        raise HTTPException(
+                status_code=HTTP_401_UNAUTHORIZED,
+                headers=path,
+                detail="Your token is incorrect. Please log in again")
+
+
+async def is_logged_in(
+    request: Request, db: Session = Depends(get_db),
+        jwt: str = Depends(get_authorization_cookie)) -> bool:
+    """
+    A dependency function protecting routes for only logged in user
+    """
+    await check_jwt_token(db, jwt)
+    return True
+
+
+async def logged_in_user(
```

Comment:
This is not a verb, so it's hard to understand what the function suppose to do.

Should it be `is_logged_in`?
How does it different from the current `is_logged_in`?

Reply:
The difference is described in the Type Annotations.
They both protecting a route for only logged in user.
The difference is while is_logged_in returns a Boolean, this one returns the User object itself.
I think its extremely useful having this kind of dependency. Then you don't need to query the database in each route when you want to use the current user's different data.
At first i had just this one, but you did not want the database to be queried when the jwt is checked. So i made two different dependencies that to different actions, thinking about other developers having useful dependencies.

Reply:
So how is it different from `get_current_user`?

You must understand you're building an API for other programmers.
All the functions *must* be easily understood from their names only.

Currently it's really hard to tell what this function (or any other function in this file) do from only looking at it's name.
We must find a better naming that will describe it in a way the other programmers would be able to understand what is does.